hutch_python/cache.py

```diff
@@ -67,6 +67,9 @@ def write_file(self):
                     + body.format(datetime.datetime.now()))
             for name, obj in self.objs.__dict__.items():
                 text += '{:<20} {}\n'.format(name, obj.__class__)
+            if not db_path.exists():
+                db_path.touch()
+                db_path.chmod(0o666)
```

Comment:
Run the `chmod` outside the if statement?

Reply:
Fails if not already chmoded

Reply:
e.g. if you didn't have permission to write before, you don't have permission to chmod now